core/execution/future/market.go

```diff
@@ -240,7 +240,8 @@ func NewMarket(
 	)
 	positionEngine := positions.NewSnapshotEngine(log, positionConfig, mkt.ID, broker)
 
-	feeEngine, err := fee.New(log, feeConfig, *mkt.Fees, asset, positionFactor)
+	quoteAssetQuantum, _ := collateralEngine.GetAssetQuantum(asset)
```

Comment:
what if asset quantum changes? is this being propagated somewhere?

Reply:
that's a good point, that I ignored.
I can pass around the collateral engine instead and just GetAssetQuantum everytime?